src/components/PeoplePage.tsx

```diff
@@ -1,8 +1,100 @@
+import React, { useEffect, useState } from 'react';
+
 import { PeopleFilters } from './PeopleFilters';
-import { Loader } from './Loader';
 import { PeopleTable } from './PeopleTable';
+import { getPeople } from '../api';
+import { useFilters } from '../hooks/useFilters';
+
+import { Person } from '../types';
+
+const filteringPeople = (
+  peopleArray: Person[],
+  filters: {
+    sexInSearch: string | null;
+    query: string;
+    cents: string[];
+    sort: string | null;
+    order: string | null;
+  },
+) => {
+  let peopleArrayCopy = [...peopleArray];
+  const order = filters.order ? -1 : 1;
+
+  switch (filters.sexInSearch) {
+    case 'm':
+      peopleArrayCopy = peopleArrayCopy.filter(person => person.sex === 'm');
+      break;
+    case 'f':
+      peopleArrayCopy = peopleArrayCopy.filter(person => person.sex === 'f');
+      break;
+    default:
+      break;
+  }
+
+  switch (filters.sort) {
+    case 'name':
+      peopleArrayCopy = peopleArrayCopy.sort(
+        (person1, person2) => person1.name.localeCompare(person2.name) * order,
+      );
+      break;
+    case 'sex':
+      peopleArrayCopy = peopleArrayCopy.sort(
+        (person1, person2) => person1.sex.localeCompare(person2.sex) * order,
+      );
+      break;
+    case 'born':
+      peopleArrayCopy = peopleArrayCopy.sort(
+        (person1, person2) => (person1.born - person2.born) * order,
+      );
+      break;
+    case 'died':
+      peopleArrayCopy = peopleArrayCopy.sort(
+        (person1, person2) => (person1.died - person2.died) * order,
+      );
+      break;
+  }
+
+  if (filters.query) {
+    peopleArrayCopy = peopleArrayCopy.filter(person =>
+      person.name
+        .toLowerCase()
+        .includes(filters.query.toLowerCase().trimStart()),
+    );
+  }
+
+  if (filters.cents.length !== 0) {
+    peopleArrayCopy = peopleArrayCopy.filter(person =>
+      filters.cents.includes(`${Math.floor(person.born / 100) + 1}`),
+    );
+  }
+
+  return peopleArrayCopy;
+};
+
+export const PeoplePage: React.FC = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [fetchingError, setFetchingError] = useState(false);
+  const { sexInSearch, query, cents, sort, order } = useFilters();
+
+  useEffect(() => {
+    setIsLoading(true);
+    setFetchingError(false);
+
+    getPeople()
+      .then(fetchedPeople => setPeople(fetchedPeople))
+      .catch(() => setFetchingError(true))
+      .finally(() => setIsLoading(false));
```

Comment:
It's better to move the logic inside the useEffect hook to a separate function, as it improves code readability, reusability, and makes testing easier. It also helps keep useEffect focused on side effects.